packages/core/nuxt-theme-module/index.js

```diff
@@ -19,58 +19,78 @@ const log = {
 module.exports = async function DefaultThemeModule(moduleOptions) {
   log.info(chalk.green('Starting Theme Module'));
 
-  const agnosticThemeDir = path.join(__dirname, 'theme');
-  const projectLocalThemeDir = this.options.buildDir.replace('.nuxt', '.theme');
+  if (typeof moduleOptions.routes === 'undefined') {
+    moduleOptions.routes = true;
+  }
 
-  const agnosticThemeFiles = getAllFilesFromDir(agnosticThemeDir).filter(file => !file.includes(path.sep + 'static' + path.sep));
+  const agnosticThemeDir = path.join(__dirname, 'theme');
+  const targetDirectory = moduleOptions.generate && moduleOptions.generate.path
+    ? moduleOptions.generate.path
+    : '.theme';
 
   const compileAgnosticTemplate = (filePath) => {
     return compileTemplate(
       path.join(__dirname, filePath),
-      this.options.buildDir.split('.nuxt').pop() + '.theme' + path.sep + filePath.split('theme' + path.sep).pop(),
+      this.options.buildDir.split('.nuxt').pop() + targetDirectory + path.sep + filePath.split('theme' + path.sep).pop(),
       {
-        apiClient: moduleOptions.apiClient,
-        helpers: moduleOptions.helpers,
-        composables: moduleOptions.composables
+        generate: {
+          replace: {
+            apiClient: moduleOptions.generate.replace.apiClient,
+            composables: moduleOptions.generate.replace.composables
+          }
+        }
       });
   };
+  let projectLocalThemeDir;
+  let themeDirectoriesPaths;
+
+  if (moduleOptions.generate) {
```

Comment:
**code organization** how about externalizing al generate-related code to separate file and just using `apply` || `call` to pass the context and `moduleOptions`  (+ other required) as arguments

Reply:
I separated it and load only when `moduleOptions.generate`